src/proposals/20240417/SparkGnosis_20240417.t.sol

```diff
@@ -0,0 +1,312 @@
+    address public constant USDC             = 0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83;
+    address public constant USDC_PRICE_FEED  = 0x6FC2871B6d9A94866B7260896257Fd5b50c09900;
+    address public constant USDT             = 0x4ECaBa5870353805a9F068101A40E0f32ed605C6;
+    address public constant USDT_PRICE_FEED  = 0x6FC2871B6d9A94866B7260896257Fd5b50c09900;
+    address public constant EURE             = 0xcB444e90D8198415266c6a2724b7900fb12FC56E;
+    address public constant EURE_PRICE_FEED  = 0xab70BCB260073d036d1660201e9d5405F5829b7a;
+
+    address public constant XDAI_PRICE_FEED_OLD = 0x678df3415fc31947dA4324eC63212874be5a82f8;
+    address public constant XDAI_PRICE_FEED_NEW = 0x6FC2871B6d9A94866B7260896257Fd5b50c09900;
+
+    constructor() {
+        id = '20240417';
+    }
+
+    function setUp() public {
+        vm.createSelectFork(getChain('gnosis_chain').rpcUrl, 33352951);  // April 9, 2024
+        payload = deployPayload();
+
+        loadPoolContext(poolAddressesProviderRegistry.getAddressesProvidersList()[0]);
+    }
+
+    function test_poolUpgrade() public {
+        vm.prank(address(poolAddressesProvider));
+        assertEq(IProxyLike(payable(address(pool))).implementation(), POOL_IMPLEMENTATION_OLD);
+        assertEq(pool.FLASHLOAN_PREMIUM_TOTAL(), 0);  // This doesn't really need to be checked anymore as it was patched in code, but we have it for good measure
+
+        GovHelpers.executePayload(vm, payload, executor);
+
+        vm.prank(address(poolAddressesProvider));
+        assertEq(IProxyLike(payable(address(pool))).implementation(), POOL_IMPLEMENTATION_NEW);
+        assertEq(pool.FLASHLOAN_PREMIUM_TOTAL(), 0);
+    }
+
+    function test_collateralOnboarding() public {
+        ReserveConfig[] memory allConfigsBefore = createConfigurationSnapshot('', pool);
+
+        assertEq(allConfigsBefore.length, 4);
+
+        GovHelpers.executePayload(vm, payload, executor);
+
+        ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot('', pool);
+
+        assertEq(allConfigsAfter.length, 8);
+
+        // sxDAI
+        ReserveConfig memory sxdai = ReserveConfig({
+            symbol:                  'sDAI',
+            underlying:               SXDAI,
+            aToken:                   address(0),  // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            variableDebtToken:        address(0),  // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            stableDebtToken:          address(0),  // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            decimals:                 18,
+            ltv:                      70_00,
+            liquidationThreshold:     75_00,
+            liquidationBonus:         106_00,
+            liquidationProtocolFee:   10_00,
+            reserveFactor:            10_00,
```

Comment:
We have to pick a non-zero number for this (Config Engine prevents 0), so I'm just using 10% as a default, but it doesn't matter because borrowing is not enabled.